c-dependencies/js-compute-runtime/builtins/backend.cpp

```diff
@@ -17,6 +19,617 @@
+
+    // Cipher suites using ChaCha20.
+    aliases.insert({CHACHA20, filterByEncryption(all, std::set{Encryption::CHACHA20POLY1305})});
+
+    // Cipher suites using triple DES.
+    auto triple_des = filterByEncryption(all, std::set{Encryption::TRIPLE_DES});
+    aliases.insert({TRIPLE_DES, triple_des});
+
+    // Cipher suites using SHA1.
+    auto sha1 = filterByMessageDigest(all, std::set{MessageDigest::SHA1});
+    aliases.insert({SHA1, sha1});
+    aliases.insert({SHA, sha1});
+    // Cipher suites using SHA256.
+    aliases.insert({SHA256, filterByMessageDigest(all, std::set{MessageDigest::SHA256})});
+    // Cipher suites using SHA384.
+    aliases.insert({SHA384, filterByMessageDigest(all, std::set{MessageDigest::SHA384})});
+
+    // COMPLEMENTOFDEFAULT:
+    // The ciphers included in ALL, but not enabled by default. Currently this includes all RC4 and
+    // anonymous ciphers. Note that this rule does not cover eNULL, which is not included by ALL
+    // (use COMPLEMENTOFALL if necessary). Note that RC4 based cipher suites are not supported by
+    // Fastly and the only supported anonymous ciphers are `ecdh` and `triple_des`.
+    auto complementOfDefault(ecdh);
+    complementOfDefault.insert(complementOfDefault.end(), triple_des.begin(), triple_des.end());
+    aliases.insert({COMPLEMENTOFDEFAULT, complementOfDefault});
+
+    // The content of the default list is determined at compile time and normally corresponds to
+    // ALL:!COMPLEMENTOFDEFAULT:!eNULL.
+    aliases.insert({DEFAULT, parse("ALL:!COMPLEMENTOFDEFAULT:!eNULL")});
+  }
+
+  std::vector<Cipher> parse(std::string_view expression) {
+    /**
+     * All ciphers by their openssl alias name.
+     */
+    auto elements = splitCipherSuiteString(std::string(expression));
+    std::vector<Cipher> ciphers;
+    std::vector<Cipher> removedCiphers;
+    for (auto element : elements) {
+
+      if (element.rfind(DELETE, 0) == 0) {
+        auto alias = element.substr(1);
+        if (aliases.find(alias) != aliases.end()) {
+          remove(&aliases, ciphers, alias);
+        }
+      } else if (element.rfind(EXCLUDE, 0) == 0) {
+        auto alias = element.substr(1);
+        if (aliases.find(alias) != aliases.end()) {
+          auto toAdd = aliases[alias];
+          removedCiphers.insert(removedCiphers.end(), toAdd.begin(), toAdd.end());
+        }
+      } else if (element.rfind(TO_END, 0) == 0) {
+        auto alias = element.substr(1);
+        if (aliases.find(alias) != aliases.end()) {
+          moveToEnd(&aliases, ciphers, alias);
+        }
+      } else if ("@STRENGTH" == element) {
+        strengthSort(ciphers);
+        break;
+      } else if (aliases.find(element) != aliases.end()) {
+        add(&aliases, ciphers, element);
+      } else if (element.find(AND) != std::string::npos) {
+        auto intersections = split(element, std::string(1, AND).append("\\"));
```

Comment:
This will allocate the string "+\\" for each iteration that makes it to this branch. As the downstream consumer takes a `std::string_view`, you can instead use a literal to avoid the allocation.
```suggestion
        auto intersections = split(element, "+\\");
```